### PR TITLE
Addon-interactions: Fix duplicated elements in the panel

### PR DESCRIPTION
**Pull request by yannbf**
Issue: #19969 

## What I did

This PR fixes a logic that was broken in alpha 48 and adds E2E tests so we avoid getting into that situation again

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: code/e2e-tests/addon-interactions.spec.ts [hunk 1]
```diff
@@ -34,10 +34,75 @@ test.describe('addon-interactions', () => {
...
+    await expect(await interactionsRow.count()).toEqual(3);
+    await expect(interactionsTab).toContainText(/(3)/);
+
+    // Test remount state (from toolbar) - Interactions have rerun, count is correct and values are as expected
+    const remountComponentButton = await page.locator('[title="Remount component"]');
+    await remountComponentButton.click();
+    await interactionsRow.first().isVisible();
+    await expect(await interactionsRow.count()).toEqual(3);
+    await expect(interactionsTab).toContainText(/(3)/);
+  });
```
Comment (ndelangen) [inline code/e2e-tests/addon-interactions.spec.ts:107]: ❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️

**Review by ndelangen — APPROVED** [review]

File: code/lib/instrumenter/src/instrumenter.ts [hunk 2]
```diff
@@ -333,7 +333,8 @@ export class Instrumenter {
   // returns the original result.
   track(method: string, fn: Function, args: any[], options: Options) {
     const storyId: StoryId =
-      args?.[0]?.__storyId__ || global.window.__STORYBOOK_PREVIEW__?.urlStore?.selection?.storyId;
+      args?.[0]?.__storyId__ ||
+      global.window.__STORYBOOK_PREVIEW__.selectionStore.selection.storyId;
```
Comment (thafryer) [inline code/lib/instrumenter/src/instrumenter.ts:337]: What's the difference here between `selectionStore` and `urlStore`?

**Review by shilman — APPROVED** [review]
@yannbf Looking great. Any idea why the e2e tests are failing?

File: code/lib/instrumenter/src/instrumenter.ts (hunk 2 above)
Comment (tmeasday) [reply code/lib/instrumenter/src/instrumenter.ts:337]: We renamed the variable 😬

**Review by tmeasday — APPROVED** [review]
This is great @yannbf

File: code/lib/instrumenter/src/instrumenter.ts (hunk 2 above)
Comment (yannbf) [reply code/lib/instrumenter/src/instrumenter.ts:337]: It's the same. @tmeasday reworked the PreviewWeb interface and split/refactored some code to make it easier for React Native projects, which led to renaming `urlStore` to `selectionStore`. You can see more about it here https://github.com/storybookjs/storybook/pull/19623

**Comment by yannbf** [comment]
> @yannbf Looking great. Any idea why the e2e tests are failing?

I'll take a look at it now, from the looks of CI it seems that it only affects a few projects, so probably a bug in the clearing of the form in those frameworks

**Pull request merged**